PetaPoco/Core/PocoData.cs

```diff
@@ -174,7 +187,7 @@ public static PocoData ForObject(object obj, string primaryKeyName, IMapper defa
         public Delegate GetFactory(string sql, string connectionString, int firstColumn, int columnCount, IDataReader reader, IMapper defaultMapper)
         {
             // Create key for cache lookup
-            var key = Tuple.Create(sql, connectionString, firstColumn, columnCount);
+            var key = Tuple.Create(defaultMapper.GetType().FullName, sql, connectionString, firstColumn, columnCount);
```

Comment:
**question:** What's the use case for adding the mapper to the cache key? Under what conditions would a type being used with a given connection string potentially be used with two different mappers?

**issue:** If the mapper is needed as part of the key, we should do `var mapper = Mappers.GetMapper(Type, defaultMapper);` up here and use that mapper, not the default mapper.

Reply:
I just saw the same pattern as with `_pocoDatas` and thought I'd make all caches use the mapper as part of the key. Defensive programming.
I can't think of a use case where mapper would differ, so if you feel confident that this is overkill, I'll undo it.

Reply:
Just for context, the issue with the cache for `_pocoDatas` is one that does come up occasionally, if rarely. I think you saw it while working on the integration tests. In my case, I had one POCO class that I was using against two different databases -- one which used snake_case for columns and one which used PascalCase. This meant 2 different mappers, therefore 2 different `PocoData`s.

For this change, the SQL and connection string (= database) are already part of the cache key. Even if you've got a single POCO class used for two different tables in the same database with different mapping conventions, I think the SQL itself would be different, and therefore you'd already have two different cache keys.

Reply:
I've removed this change.